JSTask4.js

```diff
@@ -0,0 +1,227 @@
+const comp1 = [
+    {
+        first_name: "Ben",
+        last_name: "Almond",
+        age: 28,
+        position: "Senior Engineer",
+        company: "ABC",
+        location: "TVM",
+        appraisal: 0,
+    },
+    {
+        first_name: "Andrew",
+        last_name: "Lords",
+        age: 22,
+        position: "Engineer",
+        company: "ABC",
+        location: "TVM",
+        appraisal: 3,
+    },
+    {
+        first_name: "Arathi",
+        last_name: "A",
+        age: 42,
+        position: "PM",
+        company: "ABC",
+        location: "TVM",
+        appraisal: 9,
+    },
+];
+const comp2 = [
+    {
+        first_name: "Subin",
+        last_name: "Viswa",
+        age: 28,
+        position: "Engineer",
+        company: "ABC",
+        location: "Kochi",
+        appraisal: 1,
+    },
+    {
+        first_name: "Dev",
+        last_name: "Raj",
+        age: 21,
+        position: "Architect",
+        company: "ABC",
+        location: "Kochi",
+        appraisal: 2,
+    },
+    {
+        first_name: "Thomas",
+        last_name: "Mathew",
+        age: 28,
+        position: "Architect",
+        company: "ABC",
+        location: "Kochi",
+        appraisal: 5,
+    },
+    {
+        first_name: "Deepa",
+        last_name: "Thomas",
+        age: 31,
+        position: "Senior Engineer",
+        company: "ABC",
+        location: "Kochi",
+        appraisal: 4,
+    },
+    {
+        first_name: "Antony",
+        last_name: "Thomas",
+        age: 31,
+        position: "Senior Engineer",
+        company: "ABC",
+        location: "Kochi",
+        appraisal: 4,
+    },
+];
+
+// Task 1
+// Create a new array named employees by merging ‘comp1’ and ‘comp2’ using a spread operator.
+
+let merge = [...comp1, ...comp2];
+console.log(merge);
+
+
+// Task 2
+/* Iter the whole employee list and log the employee's full name by creating a function.
+(Use spread operator, rest parameters, Object destructuring, array.join) */
+
+for(let i=0;i<merge.length;i++)
+{
+    for(employee in merge[i])
+    {
+        console.log(merge[i].first_name);
+        break;
+    }
+}
+function full_name(){
+    for(let i=0;i<merge.length;i++)
+    {
+        for(names in merge[i])
+        {
+            let merge1 = merge[i].first_name + " " + merge[i].last_name;
+            console.log(merge1);
+            break;
+        }
+    }
+}
+full_name();
+
+
+// Task 3
+// List all employees having the role 'Engineer' and age above '25'.
+
+let pos1 = merge.filter(pos => pos.position.indexOf('Engineer')!==-1);
+console.log(pos1);
+let pos2 = merge.filter(yr => yr.age > 25);
+console.log(pos2);
+
+let roleage = [];
+for(i in merge)
+{
+    if(merge[i].position == 'Engineer' && merge[i].age > 25)
+    {
+        roleage.push(merge[i].first_name);
+    }
+}
+console.log(roleage);
+
```

Comment:
The result includes employees other than engineers. Update the code an filter appropriately.